assimilation_code/modules/assimilation/filter_mod.f90

```diff
@@ -7,7 +7,7 @@ module filter_mod
 !------------------------------------------------------------------------------
 use types_mod,             only : r8, i8, missing_r8, metadatalength, MAX_NUM_DOMS, MAX_FILES
 
-use options_mod,           only : get_missing_ok_status, set_missing_ok_status
+use options_mod,           only : get_missing_ok_status
```

Comment:
set_missing_ok_status() is not called by filter --  The model_mod must call it.
If the model does not call it, the default is for get_missing_ok_status to return FALSE, meaning
missing values are not allowed.

Reply:
You are right - removing `allow_missing_clm` is a non-backward compatible change. I searched the entire 'main' branch, and it is the only namelist it appears in is the example filter_mod.nml ... not in a single input.nml.  I understand that there may be one out there in the wild ...

Reply:
What is not part of this pull request is the change to the CLM input.nml and model_mod.f90.  The proposed model_mod.f90 DOES call the set_missing_ok_status. This PR is intended to parcel out the changes in smaller (easier to understand and test) portions.   By _**not**_ calling set_missing_ok_status, the options_mod has the traditional behavior in that missing values are not allowed.

Reply:
i'm fine with the change, and i'm very fine with parceling out the CLM changes in bite-sized chunks.  i'm glad to know our repo doesn't have this item in any input.nmls.  it's unlikely, but users may have older input.nmls that include allow_missing_clm in their filter namelist, so my request here is just to be sure our changelog and release notes include something about this item being removed from the filter namelist.  after updating if they get an error starting filter they'll have an idea of what's wrong.  seems like a pretty low risk change and it's moving functionality into a more appropriate place.